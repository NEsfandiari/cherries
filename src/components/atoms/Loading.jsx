import React from 'react'
import Spinner from 'react-spinkit'
import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${({ height }) => height || '0vh'};
  .sk-spinner {
    color: white;
  }
  .sk-spinner:hover {
    color: #e20031;
    transition: 0.3s ease;
  }
`

const Loading = props => {
  return (
    <Container height={props.height}>
      <Spinner name="three-bounce" />
    </Container>
  )
}

export default Loading
