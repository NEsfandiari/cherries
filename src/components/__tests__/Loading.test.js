import React from 'react'
import renderer from 'react-test-renderer'
import Loading from '../atoms/Loading'

describe('Loading', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Loading />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
