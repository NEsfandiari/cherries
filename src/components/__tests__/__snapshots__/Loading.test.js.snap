// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
<div
  className="sc-bdVaJa ckrZze"
>
  <div
    className="sk-fade-in sk-spinner sk-three-bounce"
    color="#ff0086"
    style={
      Object {
        "color": "#ff0086",
      }
    }
  >
    <div />
    <div />
    <div />
  </div>
</div>
`;
